src/Util/ReactionUtil.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * EdwardML Reactions
+ *
+ * Using the reactions system, the user can 'react' to content.  The content
+ * receiving a reaction is called the 'target'.  There are different types of
+ * reactions, and the user can select to perform any subset of them on a
+ * specific target.
+ *
+ * Key concepts:
+ *
+ * - UserRxns: a document holding the user's reaction states toward a target.
+ *
+ * - CountDoc: a document holding current reaction counts for a target.
+ *
+ * - RxnTarget: An object holding a description of the target of a reaction.
+ *       Should be obtained ONLY via functions in this file.
+ *
+ * - RxnType: An object describing a type of reaction.  Should be obtained
+ *       ONLY from const `RxnTypes` exported from this file.
+ */
+
+import { HandsClapping, Heart, Trash } from "phosphor-react";
+
+/**
+ * @typedef {Object} RxnTarget
+ * @property {"review"|"list"|"listComment"} type The type of target.
+ * @property {string} noun A noun to use in UI text.
+ * @property {string} entityId A globally unique ID for this reaction target.
+ * @property {string} countDocPath A path to the document with reaction counts.
+ * @property {string} ownerId The owner of this target.  They may be notified.
+ */
+
+/**
+ * Build a `RxnTarget` for an anime review.
+ * @param {number} animeId The id of the anime being reviewed.
+ * @param {string} reviewAuthorId The uid of the author of the reivew.
+ * @returns {RxnTarget} A `RxnTarget` for the specified anime review.
+ */
```

Comment:
Thanks! JSDoc is nice because it also gives hints in vscode, etc.  (Still not as good as typescript, since it won't alert you if you are putting the wrong type in a field or returning the wrong kind of value from a fn!)